src/jit/alloc.cpp

```diff
@@ -1,411 +1,389 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
-/*****************************************************************************/
-
 
 #include "jitpch.h"
-#ifdef _MSC_VER
+
+#if defined(_MSC_VER)
 #pragma hdrstop
-#endif
-/*****************************************************************************/
+#endif // defined(_MSC_VER)
 
-/*****************************************************************************/
-void                allocatorCodeSizeBeg(){}
-/*****************************************************************************/
-#ifdef  DEBUG
-/*****************************************************************************/
+size_t ArenaAllocator::s_defaultPageSize = 0;
+ArenaAllocator* ArenaAllocator::s_pooledAllocator;
+ArenaAllocator::MarkDescriptor ArenaAllocator::s_pooledAllocatorMark;
+LONG ArenaAllocator::s_isPooledAllocatorInUse = 0;
 
-void    __cdecl     debugStop(const char *why, ...)
+bool ArenaAllocator::useHostAllocator()
 {
-    va_list     args;
-
-    va_start(args, why);
-
-    printf("NOTIFICATION: ");
-    if  (why)
-        vprintf(why, args);
-    else
-        printf("debugStop(0)");
-
-    printf("\n");
-
-    va_end(args);
-
-    BreakIfDebuggerPresent();
+#if defined(DEBUG)
+    // When JitDirectAlloc is set, all JIT allocations requests are forwarded
+    // directly to the OS. This allows taking advantage of pageheap and other gflag
+    // knobs for ensuring that we do not have buffer overruns in the JIT.
+
+    static ConfigDWORD fJitDirectAlloc;
+    return fJitDirectAlloc.val(CLRConfig::INTERNAL_JitDirectAlloc) == 0;
+#else // defined(DEBUG)
+    return true;
```

Comment:
Shouldn't this return `false` by default? Oh, I guess you're using `host` here to mean the hosting interface, not the host OS....

My preference for something like this would be to have the method be effectively `doTheUnusualThing` instead of `doTheNormalThing` and then at the call sites the fact that we need to do something unusual in some cases jumps out.